graph/src/util/log.rs

```diff
@@ -67,5 +73,20 @@ pub fn register_panic_hook(panic_logger: Logger) {
                 );
             }
         };
+
+        // Send a shutdown signal to main which will attempt to cleanly shutdown the runtime
+        // After sending shutdown, the thread sleeps for 3 seconds then forces the process to
+        // exit because the shutdown is not always able to cleanly exit all workers
+        shutdown_sender
+            .lock()
+            .unwrap()
+            .take()
+            .expect("Shutdown signal already sent")
+            .send(())
+            .map(|_| ())
+            .map_err(|_| ())
+            .expect("Failed to signal shutdown");
```

Comment:
No particular reason to panic here, maybe there are two panics running and one of them already signaled the shutdown. In `map_err` it might be worth it to log something.